### PR TITLE
ENH: added links to signal.clean() in NiftiMasker docstring

### PR DESCRIPTION
**Pull request by alpinho**
solve issue #1940

### CODE REVIEW

**Comment by jeromedockes** [comment]
could you try with the trailing underscore for links to footnotes?  e.g. `[1]_`

http://docutils.sourceforge.net/docs/user/rst/quickref.html#footnotes

thanks!

**Comment by kchawla-pi** [comment]
Once #2096 has been merged, pull from master.

**Comment by kchawla-pi** [comment]
@jeromedockes Something weird is happenning. I checked the built page (this one is localy built. The one on CIrcleCI is adding some alphanumeric identifier to the link name), this is what I see:

image

There should be only 4 links not 5. INfact I was not expecting there to be any `(1, 2, 3,...) at all.

**Comment by jeromedockes** [comment]
> this is what I see:

I built the doc locally and see the same thing as you. maybe a sphinx issue?
anyway I think it would be best to replace these footnotes with 
```
:func:`nilearn.signal.clean`
```

which will link directly to the documentation for `clean`. also because the link in the footnote here is the url for the nilearn online documentation, whereas a relative path would be better if someone is reading the documentation elsewhere

File: nilearn/input_data/nifti_masker.py [hunk 1]
```diff
@@ -92,19 +92,19 @@ class NiftiMasker(BaseMasker, CacheMixin):
 
     detrend : boolean, optional
         This parameter is passed to signal.clean. Please see the related
-        documentation for details
+        documentation for details. [1]_
```
Comment (jeromedockes) [inline nilearn/input_data/nifti_masker.py]:
```suggestion
        documentation for details: :func:`nilearn.signal.clean`.
```

**Comment by alpinho** [comment]
@jeromedockes, I have addressed your comments accordingly. Can you revise it? @kchawla-pi @bthirion

**Comment by bthirion** [comment]
Thx !

**Pull request merged**